File: Programmers/Lv.0/배열 만들기 6/181859.java
import java.util.List;
import java.util.ArrayList;

class Solution {
    public int[] solution(int[] arr) {
        int[] answer = {};
        
        List<Integer> stkList = new ArrayList<>();
        int i = 0;
        
        while (i < arr.length) {
            if (!stkList.isEmpty() && stkList.get(stkList.size() - 1) == arr[i]) {
                stkList.remove(stkList.size() - 1);
            } else {
                stkList.add(arr[i]);
            }
            
            i++;
        }
        
        if (stkList.isEmpty()) {
            answer = new int[] {-1};
        } else {
            answer = new int[stkList.size()];
            for (int j=0; j<stkList.size(); j++) {
                answer[j] = stkList.get(j);
            }
        }
        
        return answer;
    }
}

File: Programmers/Lv.0/배열 만들기 6/181859.js
function solution(arr) {
    const answer = [];
    
    let i = 0;
    while (i < arr.length) {
        if (answer.length > 0 && answer[answer.length - 1] === arr[i]) {
            answer.splice(-1);
        } else {
            answer.push(arr[i]);
        }
        
        i++;
    }
    
    if (answer.length === 0) {
        answer.push(-1);
    }
    
    return answer;
}
